protocols/client/client.go

```diff
@@ -54,14 +56,28 @@ type dialer func(string, time.Duration) (net.Conn, error)
 //   - unix://<unix socket path>
 //   - vsock://<cid>:<port>
 //   - <unix socket path>
-func NewAgentClient(sock string, enableYamux bool) (*AgentClient, error) {
+func NewAgentClient(ctx context.Context, sock string, enableYamux bool) (*AgentClient, error) {
```

Comment:
why this change? the context is always `Background()`

Reply:
We want to be able to pass the context the runtime creates to `virtcontainers/kata_agent.go` because the runtime's context will have the trace information stored inside it so with this change, when the runtime is executed, we'll get gRPC traces for the client (runtime) side.

Reply:
make sense, thanks